### PR TITLE
Add write IO metrics and print physical written bytes in PlanNodeStats#toString

### PR DESCRIPTION
**Pull request by Yohahaha**
Summary:
Count write IO metrics when invoke WriteFile#append, and print physical written bytes when invoke printPlanWithStats.

Fixes: #10794

### CODE REVIEW

**Comment by Yohahaha** [comment]
@JkSelf @pedroerp @majetideepak would you help take a look? thanks!

**Review by JkSelf — COMMENTED** [review]
@Yohahaha Thanks for your work. Can you add a unit test to cover this change? Thanks.

File: velox/exec/PlanNodeStats.cpp [hunk 1]
```diff
@@ -95,8 +95,11 @@ std::string PlanNodeStats::toString(bool includeInputStats) const {
     }
   }
   out << "Output: " << outputRows << " rows (" << succinctBytes(outputBytes)
```
Comment (xiaoxmeng) [inline velox/exec/PlanNodeStats.cpp:97]: Do we have test to check this? Thanks!

File: velox/dwio/common/FileSink.cpp [hunk 2]
```diff
@@ -67,14 +67,19 @@ void FileSink::writeImpl(
     const std::function<uint64_t(const DataBuffer<char>&)>& callback) {
   DWIO_ENSURE(!isClosed(), "Cannot write to closed sink.");
   const uint64_t oldSize = size_;
-  for (auto& buf : buffers) {
-    // NOTE: we need to update 'size_' after each 'callback' invocation as some
-    // file sink implementation like MemorySink depends on the updated 'size_'
-    // for new write.
-    size_ += callback(buf);
+  uint64_t usec{0};
```
Comment (xiaoxmeng) [inline velox/dwio/common/FileSink.cpp:70]:
```
s/usec/writeTimeUs/
```

File: velox/connectors/hive/HiveDataSink.cpp [hunk 3]
```diff
@@ -511,10 +511,13 @@ DataSink::Stats HiveDataSink::stats() const {
   }
 
   int64_t numWrittenBytes{0};
+  int64_t writeIOTimeMicro{0};
```
Comment (xiaoxmeng) [inline velox/connectors/hive/HiveDataSink.cpp:514]:
```
s/writeIOTimeMicro/writeIOTimeUs/
```

File: velox/connectors/hive/HiveDataSink.cpp [hunk 4]
```diff
@@ -511,10 +511,13 @@ DataSink::Stats HiveDataSink::stats() const {
...
   for (const auto& ioStats : ioStats_) {
     numWrittenBytes += ioStats->rawBytesWritten();
```
Comment (xiaoxmeng) [inline velox/connectors/hive/HiveDataSink.cpp:516]: nit: can we directly add up in stats?

File: velox/connectors/Connector.h [hunk 5]
```diff
@@ -151,6 +151,7 @@ class DataSink {
   struct Stats {
     uint64_t numWrittenBytes{0};
     uint32_t numWrittenFiles{0};
+    uint64_t writeIOTimeMicro{0};
```
Comment (xiaoxmeng) [inline velox/connectors/Connector.h:154]:
```
s/writeIOTimeMicro/writeIOTimeUs/
```

File: velox/common/io/IoStatistics.h [hunk 6]
```diff
@@ -150,6 +152,7 @@ class IoStatistics {
   std::atomic<uint64_t> outputBatchSize_{0};
   std::atomic<uint64_t> rawOverreadBytes_{0};
   std::atomic<uint64_t> totalScanTime_{0};
+  std::atomic<uint64_t> writeIOTime_{0};
```
Comment (xiaoxmeng) [inline velox/common/io/IoStatistics.h:155]:
```
s/writeIOTime_/writeIoTimeUs_/
```

File: velox/common/io/IoStatistics.cpp [hunk 7]
```diff
@@ -70,6 +74,10 @@ uint64_t IoStatistics::incTotalScanTime(int64_t v) {
   return totalScanTime_.fetch_add(v, std::memory_order_relaxed);
 }
 
+uint64_t IoStatistics::incWriteIOTime(int64_t v) {
```
Comment (xiaoxmeng) [inline velox/common/io/IoStatistics.cpp:77]:
```
s/incWriteIOTime/incWriteIOTimeUs/
```

File: velox/common/io/IoStatistics.cpp [hunk 8]
```diff
@@ -46,6 +46,10 @@ uint64_t IoStatistics::totalScanTime() const {
   return totalScanTime_.load(std::memory_order_relaxed);
 }
 
+uint64_t IoStatistics::writeIOTime() const {
```
Comment (xiaoxmeng) [inline velox/common/io/IoStatistics.cpp:49]: ditto

**Review by xiaoxmeng — COMMENTED** [review]
@Yohahaha can you add a test for the new collected stats? Thanks!

**Review by Yuhta — APPROVED** [review]

**Pull request closed without merging**